iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -401,10 +402,72 @@ popo::ApplicationPortData* PoshRuntime::getMiddlewareApplication() noexcept
     }
 }
 
-popo::ConditionVariableData* getMiddlewareConditionVariable() noexcept
+cxx::expected<popo::ConditionVariableData*, MqMessageErrorType>
+PoshRuntime::requestConditionVariableFromRoudi(const MqMessage& sendBuffer) noexcept
 {
-    /// @todo implement new MqMessageType and this method
-    return nullptr;
+    MqMessage receiveBuffer;
+    if (sendRequestToRouDi(sendBuffer, receiveBuffer) && (3 == receiveBuffer.getNumberOfElements()))
+    {
+        std::string mqMessage = receiveBuffer.getElementAtIndex(0);
+
+        if (stringToMqMessageType(mqMessage.c_str()) == MqMessageType::IMPL_CONDITION_VARIABLE_ACK)
+        {
+            RelativePointer::id_t segmentId;
+            cxx::convert::fromString(receiveBuffer.getElementAtIndex(2).c_str(), segmentId);
+            RelativePointer::offset_t offset;
+            cxx::convert::fromString(receiveBuffer.getElementAtIndex(1).c_str(), offset);
+            auto ptr = RelativePointer::getPtr(segmentId, offset);
+            return cxx::success<popo::ConditionVariableData*>(reinterpret_cast<popo::ConditionVariableData*>(ptr));
+        }
+        else
+        {
+            LogError() << "Wrong response from message queue " << mqMessage;
```

Comment:
error handler here? If the runtime gets a wrong response, does it makees sense to continue or woud this be at least a SEVERE or even FATAL error?

Reply:
Since we control the data exchange over the message queue this should not happen. If it does, something has gone seriously wrong (we may have gotten the response for another request etc.). This indicates at least a MODERATE if not SEVERE error I think. FATAL I am unsure, we can somehow continue, but the message queue between some App and Roudi might be compromised.

Reply:
SEVERE will halt in debug but not in release. FATAL will terminate.

Reply:
Replaced the `assert` with and `errorHandler()` call using `SEVERE`.